packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -352,6 +357,54 @@ export function makeVirtualObjectManager(
     configurable: true,
   });
 
+  /**
+   * Assess the facetiousness of a value.  If the value is an object containing
+   * only named properties and each such property's value is a function, `obj`
+   * represents a single facet and 'one' is returned.  If each property's value
+   * is instead an object of facetiousness 'one', `obj` represents multiple
+   * facets and 'many' is returned.  In all other cases `obj` does not represent
+   * any kind of facet abstraction and 'not' is returned.
+   *
+   * @typedef {'one'|'many'|'not'} Facetiousness
+   *
+   * @param {*} obj  The (alleged) object to be assessed
+   *
+   * @returns {Facetiousness} an assessment of the facetiousness of `obj`
+   */
+  function assessFacetiousness(obj) {
+    if (typeof obj !== 'object') {
+      return 'not';
+    }
+    if (Object.getOwnPropertySymbols(obj).length !== 0) {
+      return 'not';
+    }
+    // XXX should there be some kind of prototype check here for further paranoia?
```

Comment:
Once you've determined that it should be a far object, if it is valid at all, then let the `Far` constructor take care of all further validation. This will change with far classes, and I'd like to keep it in one place.